qiita_ware/ebi.py

```diff
@@ -482,53 +403,81 @@ def generate_experiment_xml(self):
             design = ET.SubElement(experiment, 'DESIGN')
             design_description = ET.SubElement(design,
                                                'DESIGN_DESCRIPTION')
-            design_description.text = escape(clean_whitespace(
-                sample_info['prep']['experiment_design_description']))
+            edd = sample_prep.pop('experiment_design_description')
+            design_description.text = escape(clean_whitespace(edd))
             ET.SubElement(
                 design, 'SAMPLE_DESCRIPTOR', {'refname': sample_alias}
             )
 
-            self._generate_library_descriptor(
-                design, sample_name,
-                sample_info['prep']['library_construction_protocol']
-            )
+            # this is the library contruction section. The only required fields
+            # is library_construction_protocol, the other are optional
+            library_descriptor = ET.SubElement(design, 'LIBRARY_DESCRIPTOR')
+            library_name = ET.SubElement(library_descriptor, 'LIBRARY_NAME')
+            library_name.text = self._get_library_name(sample_name)
+
+            lcp = ET.SubElement(library_descriptor,
+                                "LIBRARY_CONSTRUCTION_PROTOCOL")
+            lcp.text = escape(clean_whitespace(
+                sample_prep.pop('library_construction_protocol')))
+
+            # these are not requiered field but present add them in the right
+            # format
+            if 'library_strategy' in sample_prep:
```

Comment:
There is a fair amount of code duplication in here, it can be collapsed:
```python
keys = ['library_strategy', 'library_source', 'library_selection', 'library_layout']
for k in keys:
    if k in sample_prep:
        sub_element = ET.SubElement(library_description, k.upper())
        sub_element.text = sample_prep.pop(k)
```
Also, the list `keys` can be defined as a class attribute, since it is a constant.